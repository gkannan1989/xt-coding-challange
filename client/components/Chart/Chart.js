import React from 'react';
import { LineChart } from 'react-chartkick';
import {
  NEWS_STORAGE_KEY,
  NEWS_FEED_SHOW,
  UP_ARROW_ICON,
} from '../../actions/constants';

const Chart = ({ results }) => {
  const statistics = [];

  for (let i = 0; i < results.length; i++) {
    const id = results[i].id && results[i].id.toString();

    const storage_item = localStorage.getItem(NEWS_STORAGE_KEY + id);
    const parse_storage_item = JSON.parse(storage_item);

    const item = {
      [id]: parse_storage_item != null ? parse_storage_item.vote_count : 0,
    };
    statistics.push(item);
  }
  let graph_item = JSON.stringify(statistics);
  graph_item = graph_item.replace(/[{}]/g, '');
  graph_item = graph_item.replace('[', '{');
  graph_item = graph_item.replace(']', '}');

  const data = JSON.parse(graph_item);

  return <LineChart data={data} />;
};

export default Chart;
