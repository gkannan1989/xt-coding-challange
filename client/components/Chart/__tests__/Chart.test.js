import React from 'react';
import renderer from 'react-test-renderer';
import Chart from '../Chart.js';

it('renders', () => {
  const component = renderer.create(<Chart />).toJSON();
  expect(component).toMatchSnapshot();
}); 