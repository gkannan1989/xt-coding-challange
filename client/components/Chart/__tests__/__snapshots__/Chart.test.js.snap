// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  id="chart-1"
  style={
    Object {
      "color": "#999",
      "fontFamily": "'Lucida Grande', 'Lucida Sans Unicode', Verdana, Arial, Helvetica, sans-serif",
      "fontSize": "14px",
      "height": "300px",
      "lineHeight": "300px",
      "textAlign": "center",
      "width": "100%",
    }
  }
>
  Loading...
</div>
`;
